pkg/process/config/config.go

```diff
@@ -582,6 +590,23 @@ func getHostname(ddAgentBin string) (string, error) {
 	return hostname, err
 }
 
+// getHostnameFromGRPC retrieves the hostname from the main datadog agent via GRPC
+func getHostnameFromGRPC(grpcClientFn func(ctx context.Context) (pb.AgentClient, error)) string {
+	ctx, cancel := context.WithTimeout(context.Background(), grpcAgentTimeout)
+	defer cancel()
+	if ddAgentClient, err := grpcClientFn(ctx); err == nil {
+		if reply, err := ddAgentClient.GetHostname(ctx, &pb.HostnameRequest{}); err == nil {
+			log.Debugf("Retrieved hostname:%s from datadog agent via grpc", reply.Hostname)
+			return reply.Hostname
+		}
+		log.Errorf("Cannot get hostname from datadog agent via grpc: %v", err)
+	} else {
+		log.Errorf("Cannot connect to datadog agent via grpc: %v", err)
+	}
+
+	return ""
```

Comment:
I mentioned this above, but since we are logging errors in this function a lot I think it would be good to return string and error for this function. (just a suggestion!)